tasks.py

```diff
@@ -1643,3 +1771,182 @@ def get_command_version(get_version_command: str, version_prefix: str) -> Option
+        raise Exception(
+            "Cannot detect IP family from ipaddresspool resource: no addresses found"
+        )
+
+    has_ipv4 = False
+    has_ipv6 = False
+
+    for address in addresses:
+        try:
+            # Handle IP ranges (e.g., "172.18.0.9-172.18.0.19")
+            if "-" in address:
+                start_ip, end_ip = address.split("-", 1)
+                start = ipaddress.ip_address(start_ip.strip())
+                end = ipaddress.ip_address(end_ip.strip())
+
+                # Both IPs must be same version and start <= end
+                if start.version != end.version:
+                    raise Exception(f"IP range '{address}' contains mixed IP versions")
+                if start > end:
+                    raise Exception(
+                        f"IP range '{address}' has start IP greater than end IP"
+                    )
+
+                # Use the start ip as a representative address for version checking
+                address = start
+
+            # Handle single IPs or CIDR networks
+            if is_ipv4(address):
+                has_ipv4 = True
+            elif is_ipv6(address):
+                has_ipv6 = True
+
+        except ValueError as e:
+            raise Exception(
+                f"Invalid IP address or network '{address}' found in ipaddresspool: {e}"
+            )
+
+    if has_ipv4 and has_ipv6:
+        return "dual"
+    elif has_ipv6:
+        return "ipv6"
+    elif has_ipv4:
+        return "ipv4"
+    else:
+        raise Exception(
+            "Cannot detect IP family from ipaddresspool resource: unmatched ip family addresses"
+        )
+
+
+def detect_protocol(namespace="metallb-system"):
+    """Detect protocol (BGP vs Layer2) from resource presence."""
+    bgppeer_result = run(
+        f"{kubectl_path} get bgppeer -n {namespace} "
+        f"-o jsonpath='{{.items[*].metadata.name}}'",
+        hide=True,
+        warn=True,
+    )
+    l2adv_result = run(
+        f"{kubectl_path} get l2advertisement -n {namespace} "
+        f"-o jsonpath='{{.items[*].metadata.name}}'",
+        hide=True,
+        warn=True,
+    )
+
+    if bgppeer_result.ok and bgppeer_result.stdout.strip():
+        return "bgp"
+    elif l2adv_result.ok and l2adv_result.stdout.strip():
+        return "layer2"
+    else:
+        raise Exception(
+            f"Cannot detect protocol from bgppeer or l2advertisement resources in {namespace} namespace"
+        )
+
+
+def detect_dev_env_config(cluster_name="kind"):
+    """Detect the configuration of the current dev environment."""
+
+    try:
+        config = {
+            "bgp_type": detect_bgp_type(),
+            "ip_family": detect_ip_family(),
+            "with_prometheus": detect_prometheus(),
+            "protocol": detect_protocol(),
+        }
+
+    except Exception as e:
+        raise Exit(message=f"Error: Could not auto-detect environment config: {e}")
+
+    return config
+
+
+def generate_test_filters(config):
```

Comment:
this function is redundant (it just unpacks `config`), let's just use `generate_skip` directly where needed